src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -170,35 +177,78 @@ size_t runtime::gpu::CUDNNEmitter::build_reduce_forward(const cudnnReduceTensorO
     CUDNN_SAFE_CALL(cudnnGetReductionWorkspaceSize(
         *m_ctx->cudnn_handle, desc, input_desc, output_desc, &workspace_size));
     size_t workspace_idx = allocator.reserve_workspace(workspace_size);
+
     void* alpha = m_host_parameters.allocate_by_datatype(data_type, 1.0);
     void* beta = m_host_parameters.allocate_by_datatype(data_type, 0);
 
-    // emit reduce operation
-    std::unique_ptr<gpu::primitive> reduce(
-        new gpu::primitive{[=, &desc, &input_desc, &output_desc](void** inputs, void** outputs) {
-            CUDNN_SAFE_CALL(cudnnSetReduceTensorDescriptor(desc,
-                                                           reduce_op,
-                                                           data_type,
-                                                           CUDNN_NOT_PROPAGATE_NAN,
-                                                           CUDNN_REDUCE_TENSOR_NO_INDICES,
-                                                           CUDNN_32BIT_INDICES));
+    std::unique_ptr<gpu::primitive> reduce;
+    switch (reduction_mode)
+    {
+    case ReductionMode::Reduce:
+    {
+        CUDNN_SAFE_CALL(cudnnSetReduceTensorDescriptor(desc,
+                                                       reduce_op,
+                                                       data_type,
+                                                       CUDNN_NOT_PROPAGATE_NAN,
+                                                       CUDNN_REDUCE_TENSOR_NO_INDICES,
+                                                       CUDNN_32BIT_INDICES));
+        // emit reduce operation
+        reduce.reset(new gpu::primitive{
+            [=, &desc, &input_desc, &output_desc](void** inputs, void** outputs) {
+                void* workspace_ptr = runtime::gpu::invoke_memory_primitive(m_ctx, workspace_idx);
+                CUDNN_SAFE_CALL(cudnnReduceTensor(*m_ctx->cudnn_handle,
+                                                  desc,
+                                                  nullptr,
+                                                  0,
+                                                  workspace_ptr,
+                                                  workspace_size,
+                                                  alpha,
+                                                  input_desc,
+                                                  inputs[0],
+                                                  beta,
+                                                  output_desc,
+                                                  outputs[0]));
+                debug_sync();
+            }});
+
+        break;
+    }
+
+    case ReductionMode::ArgReduce:
+    {
+        size_t indices_size = shape_size(output_shape) * sizeof(int);
```

Comment:
sizeof(int) -> sizeof(uint32_t) (to more precisely correspond to CUDNN_32BIT_INDICES)

Reply:
Resolved by adding error check as discussed